src/providers/ldap/sdap_id_op.c

```diff
@@ -354,26 +368,62 @@ static void sdap_id_conn_data_idle_handler(struct tevent_context *ev,
                                                           struct sdap_id_conn_data);
     struct sdap_id_conn_cache *conn_cache = conn_data->conn_cache;
 
-    if (conn_cache->cached_connection == conn_data) {
+    time_t now;
+    time_t idle_time;
+    int idle_timeout;
+    struct timeval tv;
+
+    if (conn_cache->cached_connection != conn_data) {
+        DEBUG(SSSDBG_TRACE_ALL, "Abandoning idle timer for released connection\n");
+        return;
+    }
+
+    now = time(NULL);
+    idle_time = conn_data->sh->idle_time;
+    idle_timeout = conn_data->sh->idle_timeout;
+
+    if (idle_time != 0 && idle_time + idle_timeout <= now) {
         DEBUG(SSSDBG_TRACE_ALL,
               "Connection has reached idle timeout, releasing it\n");
         conn_cache->cached_connection = NULL;
         sdap_id_release_conn_data(conn_data);
+        return;
+    }
+
+    memset(&tv, 0, sizeof(tv));
+    tv.tv_sec = (idle_time == 0 ? now : idle_time) + idle_timeout;
```

Comment:
Is it really required to re-start timer in case `conn_data->sh->idle_time` was already set to 0?